### PR TITLE
use serde `derive` feature rather than `serde_derive`

### LINKED ISSUE #40: Cannot generate documentation
I have been running into issues with the latest `jsonrpc v0.12.0`. Notably, I cannot generate the documentation due to these errors:

```rust
error[E0433]: failed to resolve: could not find `__private` in `_serde`
  --> /home/worleyg/.cargo/registry/src/github.com-1ecc6299db9ec823/jsonrpc-0.12.0/src/lib.rs:79:24
   |
79 | #[derive(Debug, Clone, Deserialize, Serialize)]
   |                        ^^^^^^^^^^^ could not find `__private` in `_serde`
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0433]: failed to resolve: could not find `__private` in `_serde`
  --> /home/worleyg/.cargo/registry/src/github.com-1ecc6299db9ec823/jsonrpc-0.12.0/src/lib.rs:79:37
   |
79 | #[derive(Debug, Clone, Deserialize, Serialize)]
   |                                     ^^^^^^^^^ could not find `__private` in `_serde`
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

However, I can fix this by adding `serde = { version = "1.0.119", features = ["derive"], default-features = false }` to my `Cargo.toml`:

```toml
jsonrpc = { version = "0.12.0", default-features = false }
serde = { version = "1.0.119", features = ["derive"], default-features = false }
```

It seems like `jsonrpc` doesn't enable the `derive` feature of serde, but for some reason documentation is failing to generate unless it is present. I do not know exactly why that is the case, as the crate builds fine, it is just documentation generation that fails.

### PR DESCRIPTION
**Pull request by apoelstra**
Fixes #40

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ pull your dependencies back down to unbroken versions:
 ```
 cargo update --package 'serde_json' --precise '1.0.39'
 cargo update --package 'serde_derive' --precise '1.0.98'
```
Comment (stevenroose) [inline README.md:14]: I suppose you wanted to remove this line as well..

Comment (apoelstra) [reply README.md:14]: Oh, I think we do need it but we need to first pin `serde`.

Comment (stevenroose) [reply README.md:14]: Did you check this?

Comment (apoelstra) [reply README.md:15]: Yep, checked, I was right about needing to pin `serde`. Fixed.

**Comment by tcharding** [comment]
This can be closed, was done in https://github.com/apoelstra/rust-jsonrpc/pull/61

**Pull request closed without merging**